pkg/e2e/routemonitors/routemonitors.go

```diff
@@ -237,3 +243,73 @@ func createPlot(title string) *plot.Plot {
 		plot.Label(plot.ErrorLabeler),
 	)
 }
+
+// Extracts the numbers and title from the route monitor plot files
+func (rm *RouteMonitors) ExtractData(baseDir string) error {
+	outputDirectory := filepath.Join(baseDir, "route-monitors")
+	if _, err := os.Stat(outputDirectory); os.IsNotExist(err) {
+		if err := os.Mkdir(outputDirectory, os.FileMode(0755)); err != nil {
```

Comment:
Are we 100% certain that `baseDir` already exists? If not, this can fail due to trying to create a child of a nonexistent parent. You can avoid that with `os.MkdirAll`

Reply:
Thanks for the input. I've included `os.MkdirAll`.